modApiServer/test/org/aion/api/server/ApiAionTests.java

```diff
@@ -0,0 +1,398 @@
+import org.aion.base.type.ITxReceipt;
+import org.aion.crypto.ed25519.ECKeyEd25519;
+import org.aion.evtmgr.impl.evt.EventBlock;
+import org.aion.evtmgr.impl.evt.EventDummy;
+import org.aion.evtmgr.impl.evt.EventTx;
+import org.aion.mcf.account.AccountManager;
+import org.aion.mcf.account.Keystore;
+import org.aion.zero.impl.blockchain.AionImpl;
+import org.aion.zero.impl.config.CfgAion;
+import org.aion.zero.impl.db.AionBlockStore;
+import org.aion.zero.impl.db.AionRepositoryImpl;
+import org.aion.zero.impl.types.AionBlock;
+import org.aion.zero.impl.types.AionBlockSummary;
+import org.aion.zero.types.AionTransaction;
+import org.aion.zero.types.AionTxReceipt;
+import org.junit.Test;
+
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.Assert.*;
+
+public class ApiAionTests {
+
+    private class ApiAionImpl extends ApiAion {
+
+        private boolean onBlockFlag;
+        private boolean pendingRcvdFlag;
+        private boolean pendingUpdateFlag;
+
+        @Override
+        protected void onBlock(AionBlockSummary cbs) {
+            onBlockFlag = true;
+        }
+
+        @Override
+        protected void pendingTxReceived(ITransaction _tx) {
+            pendingRcvdFlag = true;
+        }
+
+        @Override
+        protected void pendingTxUpdate(ITxReceipt _txRcpt, EventTx.STATE _state) {
+            pendingUpdateFlag = true;
+        }
+
+        public boolean allFlagsSet() {
+            return (onBlockFlag && pendingRcvdFlag && pendingUpdateFlag);
+        }
+
+
+        public ApiAionImpl(AionImpl impl) {
+            super(impl);
+            onBlockFlag = false;
+            pendingRcvdFlag = false;
+            pendingUpdateFlag = false;
+        }
+
+        public void addEvents() {
+            EventTx pendingRcvd = new EventTx(EventTx.CALLBACK.PENDINGTXRECEIVED0);
+            AionTransaction tx = new AionTransaction(null);
+            List l1 = new ArrayList<ITransaction>();
+            l1.add(tx);
+            l1.add(tx);
+            l1.add(tx);
+            pendingRcvd.setFuncArgs(Collections.singletonList(l1));
+
+            ees.add(pendingRcvd);
+
+            EventTx pendingUpdate = new EventTx(EventTx.CALLBACK.PENDINGTXUPDATE0);
+            List l2 = new ArrayList<>();
+            l2.add(new AionTxReceipt());
+            l2.add(-1);
+            pendingUpdate.setFuncArgs(l2);
+
+            ees.add(pendingUpdate);
+
+            EventBlock evBlock = new EventBlock(EventBlock.CALLBACK.ONBLOCK0);
+            AionBlockSummary abs = new AionBlockSummary(null, null, null, null);
+            evBlock.setFuncArgs(Collections.singletonList(abs));
+
+            ees.add(evBlock);
+
+            //provokes exception in EpApi.run()
+            ees.add(new EventBlock(EventBlock.CALLBACK.ONBLOCK0));
+            ees.add(new EventDummy());
+        }
+
+    }
+
+    @Test
+    public void TestCreate() {
```

Comment:
Thanks! Do you know whether we have style guidelines anywhere? I followed the notation in modAionApi's BaseAPITests class, which uses CamelCase and prints out the name each time. It would be good if we have some "official" guidelines.

Reply:
We try to follow the [Google Java Style Guide](https://google.github.io/styleguide/javaguide.html). Older code has not been brought up to date yet which is unfortunatly the case with the test cases you looked at.